app/src/main/java/org/mozilla/fenix/FenixApplication.kt

```diff
@@ -447,6 +451,21 @@ open class FenixApplication : LocaleAwareApplication(), Provider {
         }
     }
 
+    @VisibleForTesting
+    internal fun subscribeForNewAddonsIfNeeded(
+        checker: DefaultSupportedAddonsChecker,
+        installedExtensions: List<WebExtension>
+    ) {
+        val haveUnsupportedAddons = installedExtensions.any { it.isUnsupported() }
+        if (haveUnsupportedAddons) {
+            checker.registerForChecks()
+        } else {
+            // As checks are a persistent subscriptions, we have to make sure
+            // we remove any previous subscriptions.
+            checker.unregisterForChecks()
+        }
+    }
```

Comment:
@csadilek I update the PR to just check using  [WebExtension.isUnsupported()](https://github.com/mozilla-mobile/android-components/blob/5aad028635f3b0f28f90360ee814bd0b2660fc32/components/concept/engine/src/main/java/mozilla/components/concept/engine/webextension/WebExtension.kt#L473)